service/raft/raft_sys_table_storage.cc

```diff
@@ -301,8 +301,9 @@ future<> raft_sys_table_storage::execute_with_linearization_point(std::function<
     }
 }
 
-future<> raft_sys_table_storage::bootstrap(raft::configuration initial_configuation) {
-    raft::snapshot_descriptor snapshot;
+future<> raft_sys_table_storage::bootstrap(raft::configuration initial_configuation, bool nontrivial_snapshot) {
```

Comment:
Following my earlier comment, I would avoid sending a snapshot when bootstrapping a fresh, empty cluster, I think we should pass group0_upgrade_state instead here, or, if we stick to a boolean flag, rename bool nontrivial_snapshot to bool in_recovery.